fastly/fastly_test.go

```diff
@@ -164,3 +166,24 @@ BQADQQA/ugzBrjjK9jcWnDVfGHlk3icNRq0oV7Ri32z/+HQX67aRfgZu7KWdI+Ju
 Wm7DCfrPNGVwFWUQOmsPue9rZBgO
 -----END CERTIFICATE-----`
 }
+
+// escapePercentSign uses Terraform's escape syntax (i.e., repeating characters)
+// to properly escape percent signs (i.e., '%').
+//
+// There are two significant places where '%' can show up:
+// 1. Before a left curly brace (i.e., '{').
+// 2. Not before a left curly brace.
+//
+// In case #1, we have to double escape so that Terraform does not interpret Fastly's
+// configuration values as its own (e.g., https://docs.fastly.com/en/guides/custom-log-formats).
+//
+// In case #2, we only have to single escape.
+//
+// Refer to the Terraform documentation on string literals for more details:
+// https://www.terraform.io/docs/configuration/expressions.html#string-literals
+func escapePercentSign(s string) string {
+	escapeSign := strings.ReplaceAll(s, "%", "%%")
+	escapeTemplateSequence := strings.ReplaceAll(escapeSign, "%%{", "%%%%{")
+
+	return escapeTemplateSequence
+}
```

Comment:
Any reason why you didn't include the table tests you had previously in this PR? I would say they are still useful for a fiddly helper like this.

Reply:
I can add back!